Bottom: 028eeb771c3f596b66853dce3a22d93425848187
Top:    c32b7acc311b4ad3de023cac19244ee526bc1587
Author: Jason Wang <jasowang@redhat.com>
Date:   2015-06-25 16:54:07 +0800

net: introduce skb_get_swhash()

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index b57eebf..c05c8ac 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -926,6 +926,14 @@ static inline __u32 skb_get_hash(struct sk_buff *skb)
 	return skb->hash;
 }
 
+static inline __u32 skb_get_sw_hash(struct sk_buff *skb)
+{
+	if (!skb->sw_hash)
+		__skb_get_hash(skb);
+
+	return skb->hash;
+}
+
 __u32 skb_get_hash_perturb(const struct sk_buff *skb, u32 perturb);
 
 static inline __u32 skb_get_hash_raw(const struct sk_buff *skb)
